exercises/dolica_akelloegwel/ex02_oo_basics/src/ShapeSorter.cpp

```diff
@@ -0,0 +1,60 @@
+#include "ShapeSorter.h"
+
+void ShapeSorter::PrintBasedOnType(std::vector<Shape> shapes, std::string type)
+{
+	// Print shapes that have a given type
+	for (std::vector<Shape>::iterator it = shapes.begin(); it != shapes.end(); it++) 
+	{
+		if (it->GetType().compare(type) == 0)
+			std::cout << *it << std::endl;
+	}
+}
+void ShapeSorter::PrintBasedOnNSides(std::vector<Shape> shapes, int numSides)
+{
+	// Print shapes that have a given number of sides
+	for (std::vector<Shape>::iterator it = shapes.begin(); it != shapes.end(); it++)
+	{
+		if (it->GetSides() == numSides)
+			std::cout << *it << std::endl;
+	}
+}
+void ShapeSorter::SortByAreaDesc(std::vector<Shape> shapes)

```

Comment:
Your function does two things. It sorts and prints. Avoid giving functions multiple responsibilities. The name also does not imply that it's doing two things.